modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchSearchEngine.java

```diff
@@ -269,6 +274,35 @@ public synchronized void restore(long companyId, String backupName)
 		_waitForYellowStatus();
 	}
 
+	public void setAutoCreateIndex(boolean enable) {
+		if (Validator.isBlank(_indexNameBuilder.getIndexNamePrefix())) {
```

Comment:
Cant we just use _elasticsearchConfigurationWrapper.indexNamePrefix() here?

Reply:
well, can we trust that the actual index name prefix is the same one the user entered? see  https://github.com/liferay/liferay-portal/blob/3c4993e6808b10f5b3ab48cb2bcc3b0576a3376d/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/CompanyIdIndexNameBuilder.java#L61-L69

Design wise my thinking is that the configuration is holding the user input and the IndexNameBuilder service, sanitizes, processes -> produces the final output. Even if there's not much processing, that's my reasoning. 

Surely the configuration wrapper could do the sanitation too, but not sure if moving the responsibility for having a clean prefix  from the IndexNameBuilder would be a good idea.

Reply:
I see. Ya theres definitely some inconsistency there... probably should have had some input validation on the configuration (ie ElasticsearchConnectionConfigurationModelListener, don't allow capital letters in indexNamePrefix), so that user input _would_ match what the actual index name prefix is... but changing that now in theory would require upgrade processes etc, it would probably be simple but still don't really want to deal with that. 

If we keep it like this I guess a concern could be that someone in the future might use elasticsearchConfigurationWrapper.indexNamePrefix() thinking they would get the actual prefix, not realizing theres sanitation being performed somewhere else when the index is actually created... for that reason and just simplicity sake I probably would have just moved sanitation to the wrapper. At this point the config only exists to provide a value to the indexNameBuilder, theres nothing gained by the values being potentially returned differently from the two methods.

I don't know, do you have any more thoughts after considering what I just wrote? To me all of the changes just felt like over-engineering/adding complexity when it wasn't necessarily needed, but I understand better now why you did it.

Reply:
Yeah, there are maybe some differences in how we think about the design/approach here:

For me the IndexNameBuilder is clearly what its name says: a name "builder. The configuration consumer, responsible for providing the final values for both the index name and here, implicitly, its prefix, to its own consumers. For me the IndexNameBuilder is the most logical place to get the prefix from, if the index name is also provided from there.

I don't see over-engineering here, but clear responsibilities and cleaner design.  Or, what do you exactly mean by over-engineering? The code changes are very minor. The only effort here was really updating the tests, but the complexity, the  way the IndexNameBuilder was and is still partly used there...it was something to anticipate anyways.
 
I personally don't fancy the idea of a configuration wrapper to start doing magic for the user input. Also because I  don't see that as  their purpose  in the first place. The service providing the final index name should be in my opinion the place to get the final, actual prefix. Configurations are just user input. Their wrappers here are to propagate the changes.